.github/workflows/ci.yml

```diff
@@ -75,44 +76,26 @@ jobs:
         env:
           CLOUD_DATABASE: postgres://cloud_test@localhost:5432/cloud_test?sslmode=disable
 
-  build:
-    if: ${{ github.event_name == 'pull_request' || github.ref_name  == 'master' }}
-    runs-on: ubuntu-latest
+  pull_request:
+    if: ${{ github.event_name == 'pull_request' }}
     needs: [lint, test]
-    steps:
-      - name: Checkout repo
-        uses: actions/checkout@dc323e67f16fb5f7663d20ff7941f27f5809e9b6 # v3.2.0
-        with:
-          fetch-depth: 0
+    uses: ./.github/workflows/cd.yml
+    with:
+      tag: ${GITHUB_SHA:0:7}
+    secrets: inherit
 
-      - name: ci/build-docker
-        run: make build-image
-
-      - name: ci/scan-docker-security
-        uses: aquasecurity/trivy-action@9ab158e8597f3b310480b9a69402b419bc03dbd5 # v0.8.0
-        with:
-          image-ref: "mattermost/mattermost-cloud"
-          format: "table"
-          ignore-unfixed: true
-          vuln-type: "os,library"
-          severity: "CRITICAL"
-
-      - name: ci/build-docker-e2e
-        run: make build-image-e2e
+  master:
+    if: ${{ github.ref_name  == 'master' }}
```

Comment:
To make this easily reusable across projects, can we set up `master` as a variable? You can put :

```
env:
  DEFAULT_BRANCH: master
```

at the top of the file and then use is as `env.DEFAULT_BRANCH`. 

This will make easier move this same flows across projects :)

[doc ref](https://docs.github.com/en/github-ae@latest/actions/learn-github-actions/variables)